components/brave_wallet_ui/page/screens/composer_ui/select_token_modal/select_token_modal.tsx

```diff
@@ -444,159 +544,290 @@ export const SelectTokenModal = React.forwardRef<HTMLDivElement, Props>(
+                        isDisabled={checkIsNetworkOptionDisabled(network)}
+                      >
+                        <Row width='unset'>
+                          {network.chainId !== AllNetworksOption.chainId && (
+                            <CreateNetworkIcon
+                              network={network}
+                              marginRight={8}
+                              size='medium'
+                            />
+                          )}
+                          {network.chainName}
+                        </Row>
+                        {selectedNetworkFilter.chainId === network.chainId && (
+                          <Icon name='check-normal' />
+                        )}
+                      </DropdownOption>
+                    </leo-option>
+                  ))}
+                </DropdownFilter>
+              </Row>
+            )}
+          </Column>
+          <ScrollContainer
+            fullWidth={true}
+            justifyContent={emptyTokensList ? 'center' : 'flex-start'}
+          >
+            {tokenList}
+          </ScrollContainer>
+        </PopupModal>
+        {isPanel && pendingSelectedAsset && (
+          <BottomCardWrapper>
+            <BottomCard fullWidth={true}>
+              <Row
+                padding='16px 16px 0px 16px'
+                justifyContent='flex-end'
+              >
+                <CloseButton onClick={() => setPendingSelectedAsset(undefined)}>
+                  <CloseIcon />
+                </CloseButton>
+              </Row>
+              <SelectAccount
+                token={pendingSelectedAsset}
+                accounts={accountsForPendingSelectedAsset}
+                tokenBalancesRegistry={tokenBalancesRegistry}
+                spotPrice={
+                  spotPriceRegistry
+                    ? getTokenPriceFromRegistry(
+                        spotPriceRegistry,
+                        pendingSelectedAsset
+                      )
+                    : undefined
+                }
+                onSelectAccount={handleSelectAccount}
+              />
+            </BottomCard>
+          </BottomCardWrapper>
         )}
-        <ScrollContainer
-          fullWidth={true}
-          noPadding={!!showFullFlatTokenList}
-          justifyContent={emptyTokensList ? 'center' : 'flex-start'}
-        >
-          {tokensByAccount}
-        </ScrollContainer>
-      </PopupModal>
+        {isPanel && tokenDetails && (
+          <BottomCardWrapper>
+            <BottomCard fullWidth={true}>
+              <Row
+                padding='16px 16px 0px 16px'
+                justifyContent='flex-end'
+              >
+                <CloseButton onClick={() => setTokenDetails(undefined)}>
+                  <CloseIcon />
+                </CloseButton>
+              </Row>
+              <TokenDetails token={tokenDetails} />
+            </BottomCard>
+          </BottomCardWrapper>
+        )}
```

Comment:
Do you think it would be worth it to create a generic PopupOrBottomSheet component?

Reply:
Sure! Moved to a reusable component!